### PR TITLE
[fixBug] yarnUtil restRequest error,illegal character in RM url bug

### LINKED ISSUE #3379: [Bug] yarnUtil restRequest error,illegal character in RM url
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### Java Version

1.8

### Scala Version

2.12.x

### StreamPark Version

2.1.1

### Flink Version

1.16

### deploy mode

yarn-application

### What happened

1、restart hadoop
2、start flink app on streampark
3、streampark  Unable to Capturethe normal state of the application,Actualy app is runing,But on streampark, it shows starting

### Error Exception

```log
yarnUtil restRequest error,illegal character in RM url
```

### Screenshots

微信图片_20231206201733
微信图片_20231206201741

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!(您是否要贡献这个PR?)

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by Mortal-dx**
## What changes were proposed in this pull request

Issue Number: close #3379  

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / no)

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/watcher/FlinkAppHttpWatcher.java [hunk 1]
```diff
@@ -679,7 +678,7 @@ private Overview httpOverview(Application application) throws IOException {
         && (FlinkExecutionMode.YARN_APPLICATION == application.getFlinkExecutionMode()
             || FlinkExecutionMode.YARN_PER_JOB == application.getFlinkExecutionMode())) {
       String reqURL;
-      if (StringUtils.isBlank(application.getJobManagerUrl())) {
+      if (StringUtils.isBlank(application.getJobManagerUrl()) || ! application.getJobManagerUrl().contains("http") ) {
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/watcher/FlinkAppHttpWatcher.java:681]: Can you explain why this change? I can't get it

Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/watcher/FlinkAppHttpWatcher.java]: Check if the URL is a valid http url right? if so, We can add new method in common/Utils.scala，then call the method

```
def checkHttpURL(urlString: String) = {
    Try {
      val url = new URL(urlString)
      val connection = url.openConnection.asInstanceOf[HttpURLConnection]
      connection.setRequestMethod("HEAD")
      connection.getResponseCode == HttpURLConnection.HTTP_OK
    }.getOrElse(false)
  }
```

**Comment by caicancai** [comment]
@Dsrong Thanks for your contribution, you don't seem to have a code format

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**